Lib/zipfile.py

```diff
@@ -1112,6 +1116,20 @@ def seek(self, offset, whence=0):
             if self._decrypter is not None:
                 self._init_decrypter()
 
+        # Fast seek uncompressed unencrypted file
+        if self._fast_stored_seek_nocrc and self._compress_type == ZIP_STORED \
+                and self._decrypter == None and read_offset > 0:
+            # disable CRC checking after first seeking - it would be invalid
+            self._expected_crc = None
+            # seek actual file taking already buffered data into account
+            read_offset -= len(self._readbuffer) - self._offset
+            self._fileobj.seek(read_offset, os.SEEK_CUR)
```

Comment:
Why is `seek()` needed? Can we just update few attributes?

Reply:
I tried making it working without flushing buffers and offsets and the result was so complicated I thrown it away. I also failed to make it pass tests this way.